yb-voyager/cmd/assessMigrationCommand.go

```diff
@@ -352,15 +352,49 @@ func createMigrationAssessmentStartedEvent() *cp.MigrationAssessmentStartedEvent
 func createMigrationAssessmentCompletedEvent() *cp.MigrationAssessmentCompletedEvent {
 	ev := &cp.MigrationAssessmentCompletedEvent{}
 	initBaseSourceEvent(&ev.BaseEvent, "ASSESS MIGRATION")
-	report, err := json.Marshal(assessmentReport)
+
+	finalReport := AssessMigrationPayload{
+		AssessmentJsonReport: assessmentReport,
+		SourceSizeDetails: SourceDBSizeDetails{
+			TotalDBSize: source.DBSize,
+		},
+		TargetRecommendations: TargetSizingRecommendations{},
+		MigrationComplexity:   getMigrationComplexity(), //TODO: to figure out proper algorithm
+		ConversionIssues:      schemaAnalysisReport.Issues,
+	}
+	colocatedTables, err := assessmentReport.GetColocatedTablesRecommendation()
 	if err != nil {
-		utils.PrintAndLog("Failed to serialise the assessment report to json (ERR IGNORED): %s", err)
+		utils.PrintAndLog("Failed to get the colocated tables recommendation from assessmentReport: %v", err)
+	}
+	if assessmentReport.TableIndexStats != nil {
+		for _, stat := range *assessmentReport.TableIndexStats {
+			if stat.IsIndex {
```

Comment:
It might make sense to define these methods `getTotalIndexSize`, `getTotalTableSize` , etc on the AssessmentReport struct itself (it would be the right abstraction, and would be useful if we need these calculations somewhere else..)

Reply:
I thought of having a single loop of iterating this `TableIndexStats` for all the calculation, otherwise we have to iterate it 4 times if we have to separate it out for these, I think that might not be a good idea in case there are a lot of objects (tables and indexes). Maybe I can separate this whole calculation in separate function and return all the values?

Reply:
Made the change, to separate out this in a separate function.